app/addons/fauxton/components.react.jsx

```diff
@@ -269,12 +269,63 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
   });
 
 
+  // A super-simple replacement for window.confirm()
+  var ConfirmationModal = React.createClass({
+    propTypes: {
+      visible: React.PropTypes.bool.isRequired,
+      text: React.PropTypes.string.isRequired,
+      onClose: React.PropTypes.func.isRequired,
+      onSubmit: React.PropTypes.func.isRequired
```

Comment:
is there a reason to prefer callbacks instead of managing the state with a store?